### PR TITLE
feat(replays): move new event replay to breadcrumbs section

### LINKED ISSUE #39026: Move replay video to above breadcrumbs in the Issue Details page for an event + Replays (alpha) pill
### Solution

1. Bring the replay video to above the breadcrumbs in the Issue Details on an event, rather than having a Replay section at the bottom of this page
2. a "Replays (alpha)" pill as displayed in the mocks at the top-right corner
3. Add purple play button on video

#### Figma
https://www.figma.com/file/f9ywS3TPYGjgOvx4ff4dvY/Specs%3A-Replay-in-Issues?node-id=1%3A4650

#### Screenshot
Image
Image

This closes design ticket [#38923](https://github.com/getsentry/sentry/issues/38923).

### PR DESCRIPTION
**Pull request by danecando**
### Changes
- Removes EventReplay from eventEntries
- RRWebIntegration now stands alone at the bottom of eventEntries
- Moves EventReplay into Breadcrumbs event entry
- Adds test case for rendering EventReplay as part of Breadcrumbs
- Fixes a bug with ReplayReaderParams stub

### Notes
Same as before just noting that as is in this PR a replay will not be rendered if there is a replayId but no breadcrumbs. I have seen a couple of cases of this on events during testing. Might we want to still render it in eventEntries if there are no breadcrumbs and a replayId?

### Checklist
- [x] Breadcrumb has no changes from previous style and behavior when replayId is undefined
- [x] Checked to make sure no issues were introduced to the share view
- [x] Additional props introduced to breadcrumbs do not cause incompatibilities elsewhere (breadcrumbs is only referenced in eventEntry and tests)

Closes #39026 

`EventReplay` load error state (there's still some replays that have associated replayId's that do not exist)

Screen Shot 2022-09-26 at 3 37 25 PM

Replay rendered as part of the breadcrumbs event entry

Screen Shot 2022-09-26 at 3 37 56 PM

Event rendered with no associated replayId

Screen Shot 2022-09-26 at 3 41 21 PM

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Review by ryan953 — APPROVED** [review]

**Comment by ryan953** [comment]
> a replay will not be rendered if there is a replayId but no breadcrumbs. I have seen a couple of cases of this on events during testing

Can you provide examples where this has happened? I'm curious what the situation is.

File: static/app/components/events/interfaces/breadcrumbs/index.tsx [hunk 1]
```diff
@@ -32,6 +34,9 @@ type Props = Pick<React.ComponentProps<typeof Breadcrumbs>, 'route' | 'router'>
   };
   event: Event;
   organization: Organization;
+  projectSlug: string;
+  isShare?: boolean;
+  replayId?: string;
```
Comment (ryan953) [inline static/app/components/events/interfaces/breadcrumbs/index.tsx:39]: could we read the `replayId` off the event inside here? fewer props to drill down that way.

**Comment by danecando** [comment]
> > a replay will not be rendered if there is a replayId but no breadcrumbs. I have seen a couple of cases of this on events during testing
> 
> Can you provide examples where this has happened? I'm curious what the situation is.

Here's a replay where there is a replayId but there is no breadcrumbs
https://localhost:7999/organizations/sentry-emerging-tech/issues/3632057897/?query=is%3Aunresolved

So the EventReplay doesn't end up getting shown in this example

**Comment by ryan953** [comment]
> Here's a replay where there is a replayId but there is no breadcrumbs
https://localhost:7999/organizations/sentry-emerging-tech/issues/3632057897/?query=is%3Aunresolved


In this example i'm getting a 404 trying to load the replay data (on master, not on this branch).

It looks like the error is `Failed to get ServiceWorkerRegistration objects: The user denied permission to use Service Worker.` which probably happened on pageload, before any user actions on the page. If that's the case then I wouldn't expect a replay either, because it would be less than the minimum duration and dropped.

Feels like we should watch out in case there are more situations when someone could get into a 'no breadcrumbs' situation but otherwise i'm not worried.

File: static/app/components/events/eventEntries.tsx [hunk 2]
```diff
@@ -416,13 +414,16 @@ const EventEntries = ({
...
-          projectSlug={projectSlug}
-          replayId={replayId}
+          orgId={orgSlug}
+          projectId={projectSlug}
+          renderer={children => (
+            <StyledReplayEventDataSection type="context-replay" title={t('Replay')}>
+              {children}
+            </StyledReplayEventDataSection>
+          )}
         />
```
Comment (ryan953) [inline static/app/components/events/eventEntries.tsx:427]: ugh, i just realized... can people get both the old rrweb integration and Replay rendered at the same time? the code before didn't let that happen.

**Review by ryan953 — APPROVED** [review]

**Pull request merged**